hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java

```diff
@@ -41,38 +48,45 @@ public NodeDeleteHandler() {
     }
 
     @Override
-    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
-        // implement
-    }
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {}
 
     @Override
-    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
-        requireNonNull(context);
-
-        final var op = context.body().nodeDelete();
-        throw new UnsupportedOperationException("need implementation");
-    }
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {}
 
     /**
-     * Given the appropriate context, deletes a topic.
+     * Given the appropriate context, deletes a node.
      *
      * @param context the {@link HandleContext} of the active transaction
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     @Override
     public void handle(@NonNull final HandleContext context) {
-        requireNonNull(context, "The argument 'context' must not be null");
+        requireNonNull(context);
+
+        final NodeDeleteTransactionBody transactionBody = context.body().nodeDeleteOrThrow();
+        var nodeId = transactionBody.nodeId();
+
+        validateFalse(nodeId <= 0, INVALID_NODE_ID);
```

Comment:
This check should be in pureChecks(). nodeId = 0 is valid.

Reply:
moved to purechek